flake.nix

```diff
@@ -1,50 +1,290 @@
+    cardano-mainnet-mirror.flake = false;
+
+    # cicero
+    tullia.url = "github:input-output-hk/tullia";
+
+    # non-flake nix compatibility
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
+  };
+
+  outputs = inputs: let
+    supportedSystems = [
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-linux"
+      "aarch64-darwin"
+    ];
+  in
+    inputs.flake-utils.lib.eachSystem supportedSystems (
+      system: let
+        # setup our nixpkgs with the haskell.nix overlays, and the iohk-nix
+        # overlays...
+        nixpkgs = import inputs.nixpkgs {
+          overlays = [
+            # our local packages.
+            (import ./nix/pkgs.nix)
+            # iohkNix.overlays.crypto provide libsodium-vrf, libblst and libsecp256k1.
+            inputs.iohkNix.overlays.crypto
+            # haskellNix.overlay can be configured by later overlays, so need to come before them.
+            inputs.haskellNix.overlay
+            # configure haskell.nix to use iohk-nix crypto librairies.
+            inputs.iohkNix.overlays.haskell-nix-crypto
+          ];
+          inherit system;
+          inherit (inputs.haskellNix) config;
+        };
+        inherit (nixpkgs) lib;
+
+        # We use cabalProject' to ensure we don't build the plan for
+        # all systems.
+        cabalProject = nixpkgs.haskell-nix.cabalProject' {
+          src = ./.;
+          name = "cardano-ledger";
+          compiler-nix-name = lib.mkDefault "ghc928";
+
+          # CHaP input map, so we can find CHaP packages (needs to be more
+          # recent than the index-state we set!). Can be updated with
+          #
+          #  nix flake lock --update-input CHaP
+          #
+          inputMap = {
+            "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
+          };
+
+          # tools we want in our shell, from hackage
+          shell.tools = {
+            cabal = "3.10.1.0";
+            ghcid = "0.8.8";
+            haskell-language-server = "latest";
+            hlint = {};
+            fourmolu = "0.12.0.0";
+          };
+          # and from nixpkgs or other inputs
+          shell.nativeBuildInputs = with nixpkgs;
+            [
+              (python3.withPackages (ps: with ps; [sphinx sphinx_rtd_theme recommonmark sphinx-markdown-tables sphinxemoji]))
+              haskellPackages.implicit-hie
+            ]
+            ++ (with inputs.tullia.packages.${system}; [
+              tullia
+              nix-systems
+            ]);
+          # disable Hoogle until someone request it
+          shell.withHoogle = false;
+
+          # package customizations as needed. Where cabal.project is not
+          # specific enough, or doesn't allow setting these.
+          modules = [
+            ({pkgs, ...}: {
+              packages.byron-spec-chain.configureFlags = ["--ghc-option=-Werror"];
```

Comment:
I wonder if this (from haskell.nix) is no longer necessary:
`        packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [ [ pkgs.libsodium-vrf pkgs.secp256k1 ] ];
`
And the same for  ` packages.cardano-crypto-praos.`

Reply:
yes it is no longer necessary, that configuration is now global via the `haskell-nix-crypto` overlay: https://github.com/input-output-hk/iohk-nix/blob/02f42375ee5c2bab1640f14c6389b7e91bbfec8b/overlays/haskell-nix-crypto/default.nix#L9-L13

Reply:
Good to know, thank you